optuna/notebooks/jupytercon_demo/RAPIDS_xfeat_Optuna-GPU.ipynb

```diff
@@ -0,0 +1,1803 @@
+    "            # see https://github.com/optuna/optuna/issues/1340\n",
+    "            # see https://github.com/mlflow/mlflow/issues/2931\n",
+    "            max_mlflow_tag_length = 5000\n",
+    "            for key, value in tags.items():\n",
+    "                value = str(value)  # make sure it is a string\n",
+    "                if len(value) > max_mlflow_tag_length:\n",
+    "                    tags[key] = textwrap.shorten(value, max_mlflow_tag_length)\n",
+    "\n",
+    "            mlflow.set_tags(tags) "
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Feature Engineering\n",
+    "\n",
+    "The following functions are defined to perform a few feature engineering tasks on the data. The feature_engineering function is called on the dataframe df, in this function we use cuML's [LabelEncoder](https://docs.rapids.ai/api/cuml/stable/api.html#cuml.preprocessing.LabelEncoder.LabelEncoder) and [TargetEncoder](https://docs.rapids.ai/api/cuml/stable/api.html#cuml.preprocessing.TargetEncoder) to encode the categorical columns. Arithmetic Combinations on the numerical columns using xfeat. The Arithmetic Operation performed here is addition on two columns at once. We specify the operator and r, which is used to indicate how many columns need to be combined.\n",
+    "\n",
+    "You'll also notice we use `Pipeline` from xfeat to combine two or more feature engineering tasks together. This is useful to concatenate encoders sequentially.\n",
+    "\n",
+    "Read more about Feature Encoding and Pipelining with xfeat [here](https://github.com/pfnet-research/xfeat/blob/master/_docs/feature_encoding.md)."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 4,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "def feature_engineering(df):\n",
+    "    \"\"\"\n",
+    "    Perform feature engineering and return a new df with engineered features\n",
+    "    \"\"\"\n",
+    "    df_train, df_test, y_train, y_test = train_test_split(df, TARGET_COL,random_state=np.random.seed(0), shuffle=True)\n",
+    "\n",
+    "    # Xfeat's internal fold mechanism creates RangeIndex references, so we need to do an index reset on our data frames.\n",
+    "    df_train = df_train.reset_index(drop=True)\n",
+    "    df_test = df_test.reset_index(drop=True)\n",
+    "    y_train = y_train.reset_index(drop=True)\n",
+    "    y_test = y_test.reset_index(drop=True)\n",
+    "    \n",
+    "    # Need to do this to ensure we are appropriately assigning the split values\n",
+    "    df_train[TARGET_COL] = y_train\n",
+    "    df_test[TARGET_COL] = y_test\n",
+    "    \n",
+    "    for col in CAT_COLS:\n",
+    "        out_col = f'{col}_TE'\n",
+    "        lbl_enc = LabelEncoder(handle_unknown='ignore')\n",
+    "        tar_enc = TargetEncoder(n_folds=5, smooth=TARGET_ENC_SMOOTH, split_method=TARGET_ENC_SPLIT)\n",
```

Comment:
I'm really liking the simplification here since eliminating the CPU codepath... thank you! Looks great